tests/apm/monitoring-serverOpening-001.phpt

```diff
@@ -0,0 +1,44 @@
+--TEST--
+MongoDB\Driver\Monitoring\ServerOpeningEvent
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$m = create_test_manager();
+
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $serverOpened = false;
+
+    public function serverOpening(MongoDB\Driver\Monitoring\ServerOpeningEvent $event)
+    {
+        if ($this->serverOpened) {
+            return;
+        }
+
+        $this->serverOpened = true;
+        echo "- getHost() returns a string: ", is_string($event->getHost()) ? 'yes' : 'no', "\n";
+        echo "- getTopologyId() returns an ObjectId: ", ($event->getTopologyId() instanceof MongoDB\BSON\ObjectId) ? 'yes' : 'no', "\n";
```

Comment:
A separate idea to make this test more robust would be to ensure that `ServerOpeningEvent::getTopologyId()` is the same as the topology ID reported for other events. If you'd rather not address it in this PR then please make a ticket in the epic to track that idea. I think it's reasonable to put it off for now since it could be done independently once all of the events are implemented. And when doing so, we'd probably also want to switch over to using `Manager::addSubscriber()` in all of these tests. It doesn't make much difference since this PHP process only constructs a single Manager, but it's technically correct to do so (and serves as a good example for anyone that might reference this test as example code).

Reply:
Check topology ID: https://jira.mongodb.org/browse/PHPC-1958
Use Manager::addSubscriber: https://jira.mongodb.org/browse/PHPC-1959